### DISCUSSION TITLE
Add `remix-serve --immutable` to opt-in to immutable caching for everything

### DISCUSSION THREAD

**Discussion by mjackson** [remix-run/remix#3386] [Proposals]
Strategy should be:

- `public/build`
  - `immutable`
  - `max-age: 1y`
- `public`
  - `max-age: 1h` 

This is to minimize chances of things being busted in production but also provide some reasonable caching in production for other (maybe non-hashed) stuff in `public`.

Future: we should probably have a flag for `remix-serve` that dictates how `public` caching is to be done. Something like `remix-serve --immutable` if you want to opt-in to `immutable` caching for everything.

**Comment by machour** [comment]
First part is fixed by b1ce044f812a9bd47a1cffbc1e598bfb2a2e5868.
Leaving this open for the Future enhancement
